py/farm_ng/core/event_service_tool.py

```diff
@@ -0,0 +1,172 @@
+"""
+Usage:
+python -m farm_ng.core.event_service_tool uris config.json
+
+This will print a config file will all the uris currently being published by the config
+
+python -m farm_ng.core.event_service_tool config-gen config2.json
+
+This print a sample config file with a single service called test_service, and an example recorder service configuration (note with no port numbers so scripts no not to connect)
+
+
+Runs a single service from the config file
+
+python -m farm_ng.core.event_service_tool launch1 --service-config config.json --service-name test_service
+
+"""
+
+import argparse
+import sys
+import asyncio
+from farm_ng.core.event_service_pb2 import (
+    EventServiceConfigList,
+    EventServiceConfig,
+    SubscribeRequest,
+)
+from farm_ng.core.uri_pb2 import Uri
+from farm_ng.core.events_file_writer import proto_to_json_file
+from farm_ng.core.events_file_reader import proto_from_json_file
+from farm_ng.core.event_client import EventClient
+from google.protobuf.json_format import MessageToJson
+from farm_ng.core.event_service import add_service_parser, load_service_config
+from farm_ng.core.subprocess import (
+    get_subprocess_manager,
+    Subprocess,
+    SubprocessManager,
+)
+
+
+def config_gen_command(args):
+    proto_to_json_file(
+        args.output,
+        EventServiceConfigList(
+            configs=[
+                EventServiceConfig(
+                    name=args.name,
+                    host=args.host,
+                    port=args.port,
+                    subscriptions=[
+                        SubscribeRequest(
+                            uri=Uri(path="/test", query="service_name=bar"), every_n=1
```

Comment:
@ethanrublee how do you subscribe e.g to a specific stream of an oak camera? or which granularity we want in the services

Reply:
I think if we want to subscribe to a protobuf stream, like OakSyncFrame, but only receive a sub field this could be accomplished with something like
```
path: "/oak0/sync_frame"
query: "type=OakSyncFrame&include=imu_packets&service_name=oak0"
```
Then the generic EventService can use the include_field=imu_packets to drop all fields that are not include in the query string, using the for field in message.ListFields(): if field not in include_list: message.ClearField(field) and this can be specific per subscription.

Alternatively, our services can do this with the URI path, and publish both an ``/oak0/sync_frame`` and an "/oak0/imu" message.

Reply:
Haven't implemented this yet.  Suggest this is done after we have a few more tests and use cases.

Reply:
Ok the conclusion is that each service should publish its own messages in a granular fashion such that subscriptions can be data effecient and simplify the event service/client pub/sub mechanism for now.  Later we can evaluate if we want to allow for subscription to sub-fields of protobufs.